### PR TITLE
os/bluestore/BitmapFreelistManager: readability improvements

### PR DESCRIPTION
**Pull request by xiexingguo**

### CODE REVIEW

File: src/os/bluestore/BitmapFreelistManager.cc [hunk 1]
```diff
@@ -113,33 +113,32 @@ int BitmapFreelistManager::init()
 
   // load meta
   while (it->valid()) {
-    string k = it->key();
-    if (k == "bytes_per_block") {
+    if (it->key() == "bytes_per_block") {
```
Comment (liewegas) [inline src/os/bluestore/BitmapFreelistManager.cc]: key() returns a string by value, not reference, so c_str() here is a use-after-free

Comment (xiexingguo) [inline src/os/bluestore/BitmapFreelistManager.cc]: dropped this change

**Comment by xiexingguo** [comment]
retest this please

File: src/os/bluestore/BitmapFreelistManager.cc [hunk 2]
```diff
@@ -184,29 +184,29 @@ void BitmapFreelistManager::enumerate_reset()
...
 {
   const char *p = bl.c_str();
   int bits = bl.length() << 3;
   while (start < bits) {
-    int byte = start >> 3;
-    unsigned char mask = 1 << (start & 7);
-    if (p[byte] & mask) {
+    int which_byte = start / 8;
+    int which_bit = start % 8;
+    if (p[which_byte] & byte_bit_mask[which_bit]) {
```
Comment (liewegas) [inline src/os/bluestore/BitmapFreelistManager.cc]: is this really better than 1 << which_bit ?

Comment (xiexingguo) [inline src/os/bluestore/BitmapFreelistManager.cc]: I doubt.

Repushed, can you check again?

**Comment by xiexingguo** [comment]
@liewegas Any more comments?

**Pull request merged**